File: inventory-frontend/src/components/Item.js
/*
    @author Sharat C S
*/
import React, { useEffect, useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import { Container, Paper } from '@material-ui/core';
import InvenAppbar from './InvenAppbar';
import { useNavigate } from 'react-router-dom';

const useStyles = makeStyles((theme) => ({
  root: {
    '& > *': {
      margin: theme.spacing(1),
    },
  },
}));

export default function Item() {

    const nav = useNavigate();

    const paperStyle = {padding:'50px 20px', width:600, margin:"20px auto"}
    const [item_name, setItemName] = useState('');
    const [item_number, setItemNumber] = useState('');
    const [item_cost, setItemCost] = useState('');
    const [item_mrp, setItemMrp] = useState('');
    const [item_supplier, setItemSupplier] = useState('');
    
    
    const handleNumChange = (e) => {
        const regex = /^[0-9\b]+$/;
        if(e.target.value === "" || regex.test(e.target.value)) {
            setItemNumber(e.target.value);
        }
    };

    const handleCostChange = (e) => {
        const regex = /^[0-9\b]+$/;
        if(e.target.value === "" || regex.test(e.target.value)) {
            setItemCost(e.target.value);
        }
    };

    const handleMrpChange = (e) => {
        const regex = /^[0-9\b]+$/;
        if(e.target.value === "" || regex.test(e.target.value)) {
            setItemMrp(e.target.value);
        }
    };

    const handleAddItem = (e) => {
        e.preventDefault()
        const item = {item_name, item_number, item_cost, item_mrp, item_supplier}
        console.log(item)

        fetch("http://localhost:8080/item/add",{
            method:"POST",
            headers:{"Content-Type":"application/json"},
            body:JSON.stringify(item)
        }).then(() => {
            console.log("Item Added")
            nav("/")
        })
    };

    

  const classes = useStyles();

  return (
    <Container>
        <Paper elevation={3} style={paperStyle}>
            <h1 style={{color:"blue"}}>Enter Item Details</h1>
            <form className={classes.root} noValidate autoComplete="off">
                <TextField 
                    id="outlined-basic" label="Item Name" variant="outlined" fullWidth required
                    value={item_name} 
                    onChange={(e) => setItemName(e.target.value)}
                />
                <TextField 
                    id="outlined-basic" label="Quantity" variant="outlined" fullWidth required
                    inputProps={{ inputmode: 'numeric', pattern: '[0-9]*' }}
                    value={item_number}
                    onChange={(e) => handleNumChange(e)}
                />
                <TextField 
                    id="outlined-basic" label="Cost" variant="outlined" fullWidth required
                    inputProps={{ inputmode: 'numeric', pattern: '[0-9]*' }}
                    value={item_cost}
                    onChange={(e) => handleCostChange(e)}
                />
                <TextField 
                    id="outlined-basic" label="MRP" variant="outlined" fullWidth required
                    inputProps={{ inputmode: 'numeric', pattern: '[0-9]*' }}
                    value={item_mrp}
                    onChange={(e) => handleMrpChange(e)}
                />
                <TextField 
                    id="outlined-basic" label="Item Supplier" variant="outlined" fullWidth required
                    value={item_supplier}
                    onChange={(e) => setItemSupplier(e.target.value)}
                />
                <Button 
                    variant='contained'
                    color='secondary'
                    disabled = {!item_supplier}
                    onClick={handleAddItem} 
                    >
                    Add Item
                </Button>
            </form>
        </Paper>
    </Container>
  );
}

File: inventory-backend/src/main/java/com/sharat/inventory/model/Item.java
/*
 * @author Sharat C S
 */
package com.sharat.inventory.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Item {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String item_name;
    private int item_number;
    private int item_cost;
    private int item_mrp;
    private String item_supplier;
    
    
    public Item() {
    }


    public int getId() {
        return id;
    }


    public void setId(int id) {
        this.id = id;
    }


    public String getItem_name() {
        return item_name;
    }


    public void setItem_name(String item_name) {
        this.item_name = item_name;
    }


    public int getItem_number() {
        return item_number;
    }


    public void setItem_number(int item_number) {
        this.item_number = item_number;
    }

    public int getItem_cost() {
        return item_cost;
    }


    public void setItem_cost(int item_cost) {
        this.item_cost = item_cost;
    }


    public int getItem_mrp() {
        return item_mrp;
    }


    public void setItem_mrp(int item_mrp) {
        this.item_mrp = item_mrp;
    }


    public String getItem_supplier() {
        return item_supplier;
    }


    public void setItem_supplier(String item_supplier) {
        this.item_supplier = item_supplier;
    }

    

}
